chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb

```diff
@@ -0,0 +1,114 @@
+#
+# Cookbook Name:: crowbar
+# Recipe:: crowbar-upgrade
+#
+# Copyright 2013-2016, SUSE LINUX Products GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# This recipe is for executing actions that need to be done at nodes
+# as a preparation for the system upgrade. The preparation itself
+# consists of several steps which we distinguish by various attributes
+# saved in the node structure.
+
+return unless node[:platform] == "suse"
+
+if node["crowbar_wall"]["crowbar_openstack_upgrade"]
+
+  # Actions to be run last, when admin node is already new SUSE Cloud version (6)
+  # Nodes will be restarted and their system upgraded after this.
+
+  # put HA nodes into maintenance mode (TODO is it needed? It wasn't in 4-5 upgrade...)
+
+  # Stop pacemaker resources in all clusters
```

Comment:
In upgrade 4-5, we were doing this from admin server by ssh-ing to the founder nodes first, so we could be sure resouces were stopped before the next action. (https://github.com/SUSE-Cloud/suse-cloud-upgrade/blob/master/lib/suse-cloud-upgrade-4-to-5-pre#L313)

Now, doing that here from recipe, we cannot be sure which node goes first.

So is it wrong to stop pacemaker resources here? Is it enough to use maintenance mode instead of this? Or do we need to add some orchestration to ensure founder nodes are first? @vuntz ?